EnoDatabase/EnoDatabase.cs

```diff
@@ -710,11 +778,22 @@ public async Task CalculateRoundTeamServiceStates(IServiceProvider serviceProvid
             {
                 foreach (var service in services)
                 {
+                    var key2 = (service.Id, team.Id);

```

Comment:
Is there any benefit from looping over everything twice? Can't we do this up in the first loop?

Reply:
It loops twice for benchmarking reasons, to easily see the time needed only for data retrieval. Since the first loop takes up ~99% of the time spent inside the function, it does not really matter much. Also, I hope to replace the first loop with one data retrieval from a view soon.